src/main/java/com/google/flourbot/entity/EntityModule.java

```diff
@@ -0,0 +1,86 @@
+package com.google.flourbot.entity;
+
+import com.google.flourbot.entity.EntityInterface;
+import com.google.flourbot.datastorage.FirebaseDataStorage;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.flourbot.entity.action.Action;
+import com.google.flourbot.entity.action.SheetAppendAction;
+import com.google.flourbot.entity.trigger.Trigger;
+import com.google.flourbot.entity.trigger.CommandTrigger;
+import java.util.Map;
+
+import java.lang.Exception;
+
+public class EntityModule implements EntityInterface {
+    
+    private static EntityModule instance = null;
```

Comment:
DataStorage as type here and always, if you have a possibility to use interface type - always use the interface ;)

Reply:
Good point! Do you have the cast the Interface to Implementation type when calling `getDocument()`?

Reply:
Yes I think you can. Changed the type to Interface.